content/sensu-go/5.18/guides/generate-certificates.md

```diff
@@ -0,0 +1,290 @@
+
+Unqualified name | IP address | Fully qualified domain name (FQDN) | Additional names     |
+-----------------|------------|------------------------------------|----------------------|
+backend-1        | 10.0.0.1   | backend-1.example.com              | localhost, 127.0.0.1 |
+backend-2        | 10.0.0.2   | backend-2.example.com              | localhost, 127.0.0.1 |
+backend-3        | 10.0.0.3   | backend-3.example.com              | localhost, 127.0.0.1 |
+
+Note that the additional names for localhost and 127.0.0.1 are added here for convenience and not strictly required.
+
+Use these name and address details to create two `*.pem` files and one `*.csr` file for each backend:
+
+{{< highlight shell >}}
+# Value provided for the NAME variable will be used to populate the certificate's CN record
+# Values provided in the ADDRESS variable will be used to populate the certificate's SAN records
+# For systems with multiple hostnames and IP addresses, add each to the comma-delimited value of the ADDRESS variable
+export ADDRESS=localhost,127.0.0.1,10.0.0.1,backend-1
+export NAME=backend-1.example.com
+echo '{"CN":"'$NAME'","hosts":[""],"key":{"algo":"rsa","size":2048}}' | cfssl gencert -config=ca-config.json -profile="backend" -ca=ca.pem -ca-key=ca-key.pem -hostname="$ADDRESS" - | cfssljson -bare $NAME
+
+export ADDRESS=localhost,127.0.0.1,10.0.0.2,backend-2
+export NAME=backend-2.example.com
+echo '{"CN":"'$NAME'","hosts":[""],"key":{"algo":"rsa","size":2048}}' | cfssl gencert -config=ca-config.json -profile="backend" -ca=ca.pem -ca-key=ca-key.pem -hostname="$ADDRESS" - | cfssljson -bare $NAME
+
+export ADDRESS=localhost,127.0.0.1,10.0.0.3,backend-3
+export NAME=backend-3.example.com
+echo '{"CN":"'$NAME'","hosts":[""],"key":{"algo":"rsa","size":2048}}' | cfssl gencert -config=ca-config.json -profile="backend" -ca=ca.pem -ca-key=ca-key.pem -hostname="$ADDRESS" - | cfssljson -bare $NAME
+{{< /highlight >}}
+
+<a name="copy-backend-pem"></a>
+
+You should now have a set of files for each backend:
+
+filename               | description                  | required on backend?|
+-----------------------|------------------------------|---------------------|
+`ca.pem`               | Trusted CA root certificate  | {{< check >}}       |
+`backend-*.pem`        | Backend server certificate   | {{< check >}}       |
+`backend-*-key.pem`    | Backend server private key   | {{< check >}}       |
+`backend-*.csr`        | Certificate signing request  |                     |
+
+Again, make sure to copy all backend PEM files and CA root certificate to the corresponding backend system:
+
+{{< highlight shell >}}
+
+# Directory listing of /etc/sensu/tls on backend-1:
+/etc/sensu/tls/
+├── backend-1-key.pem
+├── backend-1.pem
+└── ca.pem
+{{< /highlight >}}
+
+These files should be accessible only by the `sensu` user.
```

Comment:
These files should be read and writable by only the `Sensu` user. 

Also need to test but I think it should actually be chmod 400 (readable only) which will lock it down even more.